### ISSUE TITLE
Debezium 2.7 upgrade error in SnapshotLockingMode.

### ISSUE THREAD

**Issue by subkanthi** [Altinity/clickhouse-sink-connector#644]
![image](https://github.com/Altinity/clickhouse-sink-connector/assets/17434846/debe68da-1adc-46fa-bd4e-2e2e234bec02)

![image](https://github.com/Altinity/clickhouse-sink-connector/assets/17434846/2af7874c-faff-47d6-bbc0-bafe9c28deca)

![image](https://github.com/Altinity/clickhouse-sink-connector/assets/17434846/45dfd384-b5d3-4151-8194-4ae0db22105f)

**Comment by subkanthi** [comment]
The problem is the ServiceLoader class is not able to find the implementation in bundled JAR.

```
    public SnapshotLockProvider() {
        this.snapshotLockImplementations = (List)StreamSupport.stream(ServiceLoader.load(SnapshotLock.class).spliterator(), false).collect(Collectors.toList());
    }

```
https://stackoverflow.com/questions/50256929/service-loader-not-finding-any-implementations-when-packaged-as-a-jar

**Comment by subkanthi** [comment]
Fixed in #590
